Makefile.base

```diff
@@ -44,19 +44,22 @@ OBJ := $(OBJC) $(OBJCXX) $(ASMOBJ) $(ASSMOBJ)
 DEP := $(OBJC:.o=.d) $(OBJCXX:.o=.d) $(ASSMOBJ:.o=.d)
 
 $(BINDIR)$(MODULE)/:
+	@echo "MKDIR $(BINDIR)$(MODULE)" 2>&1
 	$(AD)mkdir -p $@
+#	ls -l -d $(BINDIR)$(MODULE) 2>&1
 
 $(BINDIR)$(MODULE).a $(OBJ): | $(BINDIR)$(MODULE)/
 
 $(BINDIR)$(MODULE).a: $(OBJ) | ${DIRS:%=ALL--%}
+	@echo "ARCHIVE $(BINDIR)$(MODULE)" 2>&1
 	$(AD)$(AR) $(ARFLAGS) $@ $?
 
-
 CXXFLAGS = $(filter-out $(CXXUWFLAGS), $(CFLAGS)) $(CXXEXFLAGS)
 
 # compile and generate dependency info
 
 $(OBJC): $(BINDIR)$(MODULE)/%.o: %.c
```

Comment:
I'd prefer that over a bug in tmpfs ;)
I just did > 5 successful builds with just an added "ls $BINDIR/$MODULE" after mkdir, all successful.
Now I removed that ls, to see if it fails again. If it does, I'll try adding the order deps as you suggested and see if it helps.